doc/frequently-asked-questions.rst

```diff
@@ -449,6 +449,51 @@ following to get the ``scientific`` package build with
 
    nix-build -A haskell.packages.integer-simple.ghc802.scientific
 
+How to get an environment with GHC and LLVM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default on x86_64 LLVM is not included with GHC:
+
+.. code:: shell
+
+   $ nix-shell -p 'haskellPackages.ghcWithPackages (ps: with ps; [mtl])'
+
+   [nix-shell:~]$ ghc -fllvm Main.hs
+   <no location info>: error:
+       Warning: Couldn't figure out LLVM version!
+                Make sure you have installed LLVM 9
+   ghc: could not execute: opt
+   
+   [nix-shell:~]$ llvm-ar
+   llvm-ar: command not found
+  
+To include LLVM in GHC you have to override the expression:
+
+.. code:: shell
+
+   $ nix-shell -p '(haskellPackages.ghcWithPackages (ps: with ps; [mtl])).override { withLLVM = true; }'
+
+   [nix-shell:~]$ ghc -fllvm Main.hs
+   [1 of 1] Compiling Main             ( Main.hs, Main.o )
+   Linking Main ...
+
+However LLVM will still not be available in the environment:
+
+.. code:: shell
+
+   [nix-shell:~]$ llvm-ar
+   llvm-ar: command not found
+
+The solution is to add it as another package to the environment:
+
+.. code:: shell
+
+   $ nix-shell -p '(haskellPackages.ghcWithPackages (ps: with ps; [mtl])).override { withLLVM = true; }' 'haskellPackages.llvmPackages.llvm'
```

Comment:
```suggestion
   $ nix-shell -p '(haskellPackages.ghcWithPackages (ps: with ps; [mtl])).override { withLLVM = true; }' 'llvmPackages.llvm'
```

Reply:
Well, the idea here is that you want to use the same LLVM version as GHC is using. So this change isn't correct, this would give LLVM 7 instead of LLVM 9.

Reply:
Oh, nevermind me, I didn't know about this reexposure of llvm.

Reply:
I mean, you could use `llvm_9`, but looking up the version of LLVM for the version of GHC you're using is a pain.

Reply:
Yeah, no, I agree this is much better how you had it. I just wasn't aware of the re-exposure and thought it was a typing mistake.